src/sinks/aws_sqs.rs

```diff
@@ -0,0 +1,412 @@
+        let encoding = config.encoding;
+        let fifo = config.queue_url.ends_with(".fifo");
+        let message_group_id = match config.message_group_id {
+            Some(value) => Some(Template::try_from(value).context(TopicTemplate)?),
+            None if fifo => return Err(Box::new(BuildError::MessageGroupIdWithFifo)),
+            None => None,
+        };
+
+        let sqs = SqsSink {
+            client,
+            queue_url: config.queue_url,
+        };
+
+        let sink = request
+            .batch_sink(
+                SqsRetryLogic,
+                sqs,
+                VecBuffer::new(batch.size),
+                batch.timeout,
+                cx.acker(),
+            )
+            .sink_map_err(|error| error!(message = "Fatal sqs sink error.", %error))
+            .with_flat_map(move |event| {
+                stream::iter(encode_event(event, &encoding, message_group_id.as_ref())).map(Ok)
+            });
+
+        Ok(sink)
+    }
+}
+
+impl Service<Vec<SendMessageEntry>> for SqsSink {
+    type Response = SendMessageResult;
+    type Error = RusotoError<SendMessageError>;
+    type Future = BoxFuture<'static, Result<Self::Response, Self::Error>>;
+
+    fn poll_ready(&mut self, _cx: &mut Context) -> Poll<Result<(), Self::Error>> {
+        Poll::Ready(Ok(()))
+    }
+
+    fn call(&mut self, mut entries: Vec<SendMessageEntry>) -> Self::Future {
+        assert_eq!(entries.len(), 1, "Sending batch is not supported.");
+
+        let entry = entries.remove(0);
+        let byte_size = entry.message_body.len();
+
+        let client = self.client.clone();
+        let request = SendMessageRequest {
+            message_body: entry.message_body,
+            message_group_id: entry.message_group_id,
+            queue_url: self.queue_url.clone(),
+            ..Default::default()
+        };
+
+        Box::pin(async move {
+            client
+                .send_message(request)
+                .inspect_ok(|result| {
+                    emit!(AwsSqsEventSent {
+                        byte_size,
+                        message_id: result.message_id.as_ref()
+                    })
+                })
+                .instrument(info_span!("request"))
+                .await
+        })
+    }
+}
+
+#[derive(Debug, Clone)]
+struct SendMessageEntry {
+    message_body: String,
+    message_group_id: Option<String>,
+}
+
+impl EncodedLength for SendMessageEntry {
+    fn encoded_length(&self) -> usize {
+        self.message_body.len()
+    }
+}
+
+impl Response for SendMessageResult {}
+
+#[derive(Debug, Clone)]
+struct SqsRetryLogic;
+
+impl RetryLogic for SqsRetryLogic {
+    type Error = RusotoError<SendMessageError>;
+    type Response = SendMessageResult;
+
+    fn is_retriable_error(&self, error: &Self::Error) -> bool {
+        match error {
+            RusotoError::HttpDispatch(_) => true,
+            RusotoError::Unknown(res) if res.status.is_server_error() => true,
+            _ => false,
```

Comment:
Should we handle AWS error codes here too? Throttling and service unavailable.

It looks like rusoto does not handle this transparently like some other AWS SDKs do: https://github.com/rusoto/rusoto/issues/234

Reply:
I do not think that we can do more that other aws sinks do...
I also noted that only `aws_cloudwatch_metrics` handle `http::StatusCode::TOO_MANY_REQUESTS`. I think we should fix other aws sinks (in another PR).

Reply:
:+1: You are right. I didn't realize that the "common" AWS errors are not propagated: https://github.com/rusoto/rusoto/issues/605

We'll need to wait for that issue to do more.